translate/cloud-client/src/main/java/com/example/translate/BatchTranslateTextWithModel.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.translate;
+
+// [START translate_v3_batch_translate_text_with_model]
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.translate.v3.BatchTranslateMetadata;
+import com.google.cloud.translate.v3.BatchTranslateResponse;
+import com.google.cloud.translate.v3.BatchTranslateTextRequest;
+import com.google.cloud.translate.v3.GcsDestination;
+import com.google.cloud.translate.v3.GcsSource;
+import com.google.cloud.translate.v3.InputConfig;
+import com.google.cloud.translate.v3.LocationName;
+import com.google.cloud.translate.v3.OutputConfig;
+import com.google.cloud.translate.v3.TranslationServiceClient;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class BatchTranslateTextWithModel {
+
+  public static void batchTranslateTextWithModel()
+      throws InterruptedException, ExecutionException, IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "YOUR-PROJECT-ID";
+    // Supported Languages: https://cloud.google.com/translate/docs/languages
+    String sourceLanguage = "your-source-language";
+    String targetLanguage = "your-target-language";
+    String inputUri = "gs://your-gcs-bucket/path/to/input/file.txt";
+    String outputUri = "gs://your-gcs-bucket/path/to/results/";
+    String modelId = "YOUR-MODEL-ID";
+    batchTranslateTextWithModel(
+        projectId, sourceLanguage, targetLanguage, inputUri, outputUri, modelId);
+  }
+
+  // Batch translate text using AutoML Translation model
+  public static void batchTranslateTextWithModel(
+      String projectId,
+      String sourceLanguage,
+      String targetLanguage,
+      String inputUri,
+      String outputUri,
+      String modelId)
+      throws IOException, ExecutionException, InterruptedException {
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (TranslationServiceClient client = TranslationServiceClient.create()) {
+      // Supported Locations: `global`, [glossary location], or [model location]
+      // Glossaries must be hosted in `us-central1`
+      // Custom Models must use the same location as your model. (us-central1)
+      String location = "us-central1";
+      LocationName parent = LocationName.of(projectId, location);
+
+      GcsSource gcsSource = GcsSource.newBuilder().setInputUri(inputUri).build();
```

Comment:
nit: I think some comments for on each section might add value. Something like `// Configure the source of the file from a GCS bucket`, `// Configure the output file in a GCS bucket`, etc.